lib/tests/unit/modules/command_test.cc

```diff
@@ -0,0 +1,216 @@
+#include <catch.hpp>
+#include "root_path.hpp"
+#include "../../common/content_format.hpp"
+
+#include <boost/algorithm/string.hpp>
+
+#include <leatherman/execution/execution.hpp>
+#include <leatherman/util/scope_exit.hpp>
+
+#include <pxp-agent/configuration.hpp>
+#include <pxp-agent/modules/command.hpp>
+
+using namespace PXPAgent;
+
+namespace fs = boost::filesystem;
+namespace lth_exe = leatherman::execution;
+namespace lth_jc = leatherman::json_container;
+
+static const std::string SPOOL_DIR { std::string { PXP_AGENT_ROOT_PATH }
+                                     + "/lib/tests/resources/test_spool" };
+static const auto STORAGE = std::make_shared<ResultsStorage>(SPOOL_DIR, "0d");
+
+static void configureTest() {
+    if (!fs::exists(SPOOL_DIR) && !fs::create_directories(SPOOL_DIR)) {
+        FAIL("Failed to create the results directory");
+    }
+    Configuration::Instance().initialize(
+            [](std::vector<std::string>) {
+                return EXIT_SUCCESS;
+            });
+}
+
+static void resetTest() {
+    if (fs::exists(SPOOL_DIR)) {
+        fs::remove_all(SPOOL_DIR);
+    }
+}
+
+static ActionRequest command_request(const std::string& params_txt, std::string transaction_id = "0987") {
+    std::string command_txt {
+            (NON_BLOCKING_DATA_FORMAT % boost::io::quoted(transaction_id)
+            % "\"command\""
+            % "\"run\""
+            % params_txt
+            % "false").str()
+    };
+
+    PCPClient::ParsedChunks command_content {
+        lth_jc::JsonContainer(ENVELOPE_TXT),
+        lth_jc::JsonContainer(command_txt),
+        {},
+        0
+    };
+
+    ActionRequest request { RequestType::NonBlocking, command_content };
+    fs::path spool_path { SPOOL_DIR };
+    auto results_dir = (spool_path / request.transactionId()).string();
+    fs::create_directories(results_dir);
+    request.setResultsDir(results_dir);
+
+    return request;
+}
+
+TEST_CASE("Modules::Command", "[modules]") {
+    SECTION("can successfully instantiate") {
+        REQUIRE_NOTHROW(Modules::Command(PXP_AGENT_BIN_PATH, STORAGE));
+    }
+}
+
+TEST_CASE("Modules::Command::hasAction", "[modules]") {
+    Modules::Command mod { PXP_AGENT_BIN_PATH, STORAGE };
+
+    SECTION("correctly reports false for a nonexistent action") {
+        REQUIRE_FALSE(mod.hasAction("foo"));
+    }
+
+    SECTION("correctly reports true for a real action") {
+        REQUIRE(mod.hasAction("run"));
+    }
+}
+
+TEST_CASE("Modules::Command::callAction successfully", "[modules]") {
+    configureTest();
+    lth_util::scope_exit config_cleaner { resetTest };
+    Modules::Command mod { PXP_AGENT_BIN_PATH, STORAGE };
+
+#ifdef _WIN32
+    static const std::string echo_params { "{ \"command\": \"cmd.exe /c echo hello\" }" };
+#else
+    static const std::string echo_params { "{ \"command\": \"echo hello\" }" };
+#endif
+
+    auto request = command_request(echo_params);
+    auto response = mod.executeAction(request);
```

Comment:
It appears that `executeAction` has sufficient unit tests via `task_test.cc` (and few scattered across `module_test.cc` and `ping_test.cc`); we also cover external modules via `external_module_test.cc`.

Reply:
are you suggesting some of these test cases don't need to be here? I'm not sure how else to test this code outside of calling executeAction to do so. I will try to improve the test descriptions in general.

Reply:
@caseywilliams looking for feedback: is there a better way for me to have phrased this such that it would have been less confusing?Thank you.

Reply:
Sure, I did not consider validating `executeAction`'s exiting behavior to be part of this ticket, since that method's been around for a while